src/drivers/net/e1000.c

```diff
@@ -282,18 +297,18 @@ static int e1000_open(struct net_device *dev) {
 	mdelay(MDELAY);
 	REG_ORIN(e1000_reg(dev, E1000_REG_RCTL),  E1000_REG_RCTL_MPE);
 
-	for (size_t i = 0; i < E1000_RXDESC_NR; i ++) {
-	        struct sk_buff *skb;
-		skb = skb_alloc(ETH_FRAME_LEN);
-		if (skb == NULL)
-			panic("Can't allocate skb %d", i);
-		rx_skbs[i]= skb;
-		rx_descs[i].buffer_address = (uint32_t) skb->mac.raw;
+	for (int i = 0; i < E1000_RXDESC_NR; i ++) {
+	        struct sk_buff *skb = skb_alloc(E1000_MAX_RX_LEN);
+		if (skb == NULL) {
+			e1000_stop(dev);
+			return -ENOMEM;
+		}
+		nic_priv->rx_skbs[i] = skb;
+		nic_priv->rx_descs[i].buffer_address = (uint32_t) skb->mac.raw;
 	}
```

Comment:
Just for the memories: it was suggested to move the buffer allocation logic to the beginning of the `e1000_open`, before setting up any hardware registers.
It is also a bit confusing that in case of an allocation error, `e1000_open` calls `e1000_stop`, which in turn also touches some registers. `E1000_REG_CTRL_RST` may be harmless though, I'm not sure about that.